### PR TITLE
Migrate Agent Key Polling to Authd module

### PR DESCRIPTION
**Pull request by juanmanuelug**
## Description
|Related issue|
|---|
|[#4422](https://github.com/wazuh/wazuh-documentation/issues/4422)|

Hi team,

With the port of the Agent Key Polling module to the Authd module, Wazuh´s needs to be updated to remove the outdated references to the Key polling module and add the needed references to the Authd documentation.

The main points to cover include:

- [x] Modify the description, the general diagram and the section [How it works](https://documentation.wazuh.com/current/user-manual/capabilities/agent-key-polling.html).
- [x] Remove the [agent-key-polling references in wazuh-modulesd](https://documentation.wazuh.com/current/user-manual/reference/daemons/wazuh-modulesd.html) and add them in authd.
- [x] Modify the [configuration](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/wodle-agent-key-polling.html) with the one added in authd.
- [x] Remove agent-key-polling [configuration reference](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/index.html) .
- [x] Updated redirects.js

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns.

### CODE REVIEW

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]

**Review by alromeros — APPROVED** [review]
LGTM!

**Comment by chemamartinez** [comment]
Do not merge this pull request until `develop` is referring to version 4.4.

**Comment by chemamartinez** [comment]
The core development was already merged https://github.com/wazuh/wazuh/pull/10865.

File: source/user-manual/agents/key-request.rst [hunk 1]
```diff
@@ -18,39 +22,41 @@ Wazuh module that allows fetching the keys stored on an external database.
...
 
 When polling by ID, the manager will retrieve the agent key by querying its ID. So the input parameters that the program will receive are as the following:
 
 ::
 
-  ./agent_key_pull.py id 001
+  ./agent_key_request.py id 001
 
-When polling by IP address, the manager will retrieve the agent key by querying his IP address, so the input parameters that the program will receive are as the following:
+When fetching by IP address, the manager will retrieve the agent key by querying his IP address. So the input parameters that the program will receive are as the following:
```
Comment (s-ocando) [inline source/user-manual/agents/key-request.rst:52]:
```suggestion
When fetching by IP address, the manager retrieves the agent key by querying his IP address. The input parameters that the program receives are as the following:
```

File: source/user-manual/agents/key-request.rst [hunk 2]
```diff
@@ -18,39 +22,41 @@ Wazuh module that allows fetching the keys stored on an external database.
...
 
-If the ``socket`` tag is not specified in the configuration block, the key polling module will call the executable with the following parameters, depending on the polling type:
+If the ``socket`` tag is not specified in the configuration block, the key request module will call the executable with the following parameters, depending on the fetching type:
 
-- Poll agent by ID
-- Poll agent by IP
+- Fetch agent information by ID
+- Fetch agent information by IP
 
 When polling by ID, the manager will retrieve the agent key by querying its ID. So the input parameters that the program will receive are as the following:
```
Comment (s-ocando) [inline source/user-manual/agents/key-request.rst:46]:
```suggestion
When polling by ID, the manager retrieves the agent key by querying its ID. The input parameters that the program receives are as the following:
```

File: source/user-manual/agents/key-request.rst [hunk 3]
```diff
@@ -59,14 +65,14 @@ When polling by IP address, the manager will retrieve the agent key by querying
 
 When the ``socket`` tag is specified the module will send the parameters through the specified socket and read the response. The performance improvement over executing the program, as it is explained above, is significant.
 
-The format in which the program will receive the data is ``option:value``, where option can be ``id`` or ``ip`` depending on the polling type.
+The format in which the program will receive the data is ``option:value``, where option can be ``id`` or ``ip`` depending on the fetching type.
```
Comment (s-ocando) [inline source/user-manual/agents/key-request.rst:68]:
```suggestion
The format in which the program receives the data is ``option:value``, where option can be ``id`` or ``ip`` depending on the fetching type.
```

**Review by s-ocando — COMMENTED** [review]
Please review the following suggestions.

**Review by s-ocando — APPROVED** [review]

**Pull request merged**